app/javascript/lib/scp-api-metrics.js

```diff
@@ -217,11 +217,25 @@ export function getLogPlotProps() {
 }
 
 /**
- * Log when a download is authorized.
+ * Extract the number of Azul files that have been chosen for download
+ */
+function getNumAzulFiles(azulFiles) {
+  let numAzulFilesChosen = 0
+  for (const val of Object.entries(azulFiles)) {
+    if (val[1].length > 0) {
+      numAzulFilesChosen += val[1].length
+    }
+  }
+  return numAzulFilesChosen
```

Comment:
I'm not sure why you need the 'if' in this method, if the length is 0 it's okay to add it.  Also, array destructuring here would help as implicit documentation to clarify what the structure of the `azulFiles` object is.
If you want to get fancy, you can also use reduce to avoid the initial set to zero line

```suggestion
  return Object.entries(azulFiles).reduce((totalNum, [azulStudyId, fileList]) =>  {
     return totalNum += fileList.length
  }, 0)
```

Reply:
I made an update based on your feedback, please let me know what you think.
I removed the extraneous if-statement and added an array destructure so the format of the object is more obvious.

I am hesitant to use reduce here though since the for-loop works and is a more generally recognizable pattern rather than `reduce` which is more explicitly JS-y and doesn't add too much in this case.